test/constants/test_constants.f90

```diff
@@ -0,0 +1,382 @@
+                             SPEED_OF_LIGHT_IN_VACUUM,&
+                             STANDARD_ACCELERATION_OF_GRAVITY
+    implicit none
+    private
+    real(dp) :: fac
+    public :: collect_constants
+
+contains
+
+!> Collect all exported unit tests
+subroutine collect_constants(testsuite)
+    !> Collection of tests
+    type(unittest_type), allocatable, intent(out) :: testsuite(:)
+    testsuite = [new_unittest("YEAR", test_year), &
+                 new_unittest("ALPHA_PARTICLE_ELECTRON_MASS_RATIO", test_ALPHA_PARTICLE_ELECTRON_MASS_RATIO),&
+                 new_unittest("ALPHA_PARTICLE_MASS", test_ALPHA_PARTICLE_MASS),&
+                 new_unittest("ATOMIC_MASS_CONSTANT", test_ATOMIC_MASS_CONSTANT),&
+                 new_unittest("AVOGADRO_CONSTANT", test_AVOGADRO_CONSTANT),&
+                 new_unittest("BOLTZMANN_CONSTANT", test_BOLTZMANN_CONSTANT),&
+                 new_unittest("ELECTRON_VOLT", test_ELECTRON_VOLT),&
+                 new_unittest("ELEMENTARY_CHARGE", test_ELEMENTARY_CHARGE),&
+                 new_unittest("FARADAY_CONSTANT", test_FARADAY_CONSTANT),&
+                 new_unittest("MOLAR_MASS_CONSTANT", test_MOLAR_MASS_CONSTANT),&
+                 new_unittest("MOLAR_VOLUME_OF_IDEAL_GAS__273_15K__101_325_KPA", test_MOLAR_VOLUME_NTP),&
+                 new_unittest("PLANCK_CONSTANT", test_PLANCK_CONSTANT),&
+                 new_unittest("SPEED_OF_LIGHT_IN_VACUUM", test_SPEED_OF_LIGHT),&
+                 new_unittest("STANDARD_ACCELERATION_OF_GRAVITY", test_STANDARD_ACCELERATION_OF_GRAVITY),&
+                 new_unittest("U_ALPHA_PARTICLE_ELECTRON_MASS_RATIO", test_U_ALPHA_PARTICLE_ELECTRON_MASS_RATIO),&
+                 new_unittest("U_ALPHA_PARTICLE_MASS", test_U_ALPHA_PARTICLE_MASS),&
+                 new_unittest("U_ATOMIC_MASS_CONSTANT", test_U_ATOMIC_MASS_CONSTANT),&
+                 new_unittest("U_AVOGADRO_CONSTANT", test_U_AVOGADRO_CONSTANT),&
+                 new_unittest("U_BOLTZMANN_CONSTANT", test_U_BOLTZMANN_CONSTANT),&
+                 new_unittest("U_ELECTRON_VOLT", test_U_ELECTRON_VOLT),&
+                 new_unittest("U_ELEMENTARY_CHARGE", test_U_ELEMENTARY_CHARGE),&
+                 new_unittest("U_FARADAY_CONSTANT", test_U_FARADAY_CONSTANT),&
+                 new_unittest("U_MOLAR_MASS_CONSTANT", test_U_MOLAR_MASS_CONSTANT),&
+                 new_unittest("U_MOLAR_VOLUME_OF_IDEAL_GAS__273_15K__101_325_KPA", test_U_MOLAR_VOLUME_NTP),&
+                 new_unittest("U_PLANCK_CONSTANT", test_U_PLANCK_CONSTANT),&
+                 new_unittest("U_SPEED_OF_LIGHT_IN_VACUUM", test_U_SPEED_OF_LIGHT),&
+                 new_unittest("U_STANDARD_ACCELERATION_OF_GRAVITY", test_U_STANDARD_ACCELERATION_OF_GRAVITY)]
+end subroutine
+
+subroutine test_year(error)
+    type(error_type), allocatable, intent(out) :: error
+
+    integer(int32) :: value, expected, diff
+
+    expected = 2022
+    value = YEAR
+    diff = expected - value
+    call check(error, diff, 0)
```

Comment:
I would recommend to do the comparison between the values directly here instead of using the precomputed difference. The error message from the test framework can be more meaningful this way.

Reply:
No problem. I'll  do the comparison directly as also suggested by @jvdp1 .